packages/core/core/src/ConfigLoader.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+import ConfigResolver from './ConfigResolver';
+import {breakStatement} from 'babel-types';
+import {conditionalExpression} from '@babel/types';
+
+export default class ConfigLoader {
+  constructor(options) {
+    this.options = options;
+  }
+
+  load(configRequest) {
+    if (configRequest.configType === 'parcel') {
+      return this.loadParcelConfig(configRequest);
+    }
+
+    return this.loadThirdPartyConfig(configRequest);
+  }
+
+  async loadParcelConfig(configRequest) {
+    let {filePath} = configRequest;
+    let configResolver = new ConfigResolver();
+
+    // Resolve plugins from cwd when a config is passed programmatically
+    let config = this.options.config
+      ? await configResolver.create({
+          ...this.options.config,
+          resolveFrom: this.options.cwd
+        })
+      : await configResolver.resolve(filePath);
+    if (!config && this.options.defaultConfig) {
+      config = await configResolver.create({
+        ...this.options.defaultConfig,
+        resolveFrom: this.options.cwd
+      });
+    }
+
+    if (!config) {
+      throw new Error('Could not find a .parcelrc');
+    }
+
+    this.parcelConfig = config;
+
+    let devDeps = [];
+    switch (configRequest.meta.actionType) {
+      case 'transformer_request':
+        devDeps = config.getTransformerNames(filePath);
+        break;
+      case 'dependency':
+        devDeps = config.getResolverNames();
+        break;
+    }
+    let devDepRequests = devDeps.map(devDep => ({
+      moduleSpecifier: devDep,
+      resolveFrom: config.resolveFrom // TODO: resolveFrom should be nearest package boundary
+    }));
+
+    let invalidations = [
```

Comment:
I wonder if something can be generalized from this. Aside from the glob pattern, will this be the same for every type of config? If config is changed or removed, then we always need to invalidate.

Reply:
This initial invalidations array will look very similar for most config yeah. I was kind of thinking there would be a Config class kind of like asset, so you could probably build up this initial array with arguments to the constructor The class would also allow you to dynamically define more invalidations and dev deps. I'm pretty sure that's why I renamed `Config.js` to `ParcelConfig.js`, but I decided to just return simple data structures for now to see how it would shape up.